torchrl/modules/tensordict_module/probabilistic.py

```diff
@@ -77,53 +77,54 @@ class of interest, e.g. :obj:`"loc"` and :obj:`"scale"` for the Normal distribut
             Default is 1000
 
     Examples:
-        >>> import functorch
         >>> import torch
         >>> from tensordict import TensorDict
-        >>> from torchrl.data import NdUnboundedContinuousTensorSpec
-        >>> from torchrl.modules import SafeProbabilisticModule, TanhNormal, NormalParamWrapper
+        >>> from tensordict.nn.functional_modules import make_functional
+        >>> from torchrl.data import CompositeSpec, NdUnboundedContinuousTensorSpec
+        >>> from torchrl.modules import (
+                NormalParamWrapper,
+                SafeModule,
+                SafeProbabilisticModule,
+                TanhNormal,
+            )
         >>> td = TensorDict({"input": torch.randn(3, 4), "hidden": torch.randn(3, 8)}, [3,])
-        >>> spec = NdUnboundedContinuousTensorSpec(4)
+        >>> spec = CompositeSpec(action=NdUnboundedContinuousTensorSpec(4), loc=None, scale=None)
         >>> net = NormalParamWrapper(torch.nn.GRUCell(4, 8))
-        >>> fnet, params, buffers = functorch.make_functional_with_buffers(net)
-        >>> module = SafeModule(fnet, in_keys=["input", "hidden"], out_keys=["loc", "scale"])
+        >>> module = SafeModule(net, in_keys=["input", "hidden"], out_keys=["loc", "scale"])
         >>> td_module = SafeProbabilisticModule(
         ...    module=module,
         ...    spec=spec,
         ...    dist_in_keys=["loc", "scale"],
         ...    sample_out_key=["action"],
         ...    distribution_class=TanhNormal,
-        ...    return_log_prob=True,
```

Comment:
I disabled this for the moment because when set to `True` I get a strange error that I've yet to get to the bottom of.

Reply:
Here's a recreation. If I set `return_log_prob=True` then my interpreted crashes and I get a "bus error" 😕 

```python
import torch
from functorch import vmap
from tensordict import TensorDict
from tensordict.nn.functional_modules import make_functional
from torchrl.data import CompositeSpec, NdUnboundedContinuousTensorSpec
from torchrl.modules import (
    NormalParamWrapper,
    SafeModule,
    SafeProbabilisticModule,
    TanhNormal,
)

td = TensorDict({"input": torch.randn(3, 4), "hidden": torch.randn(3, 8)}, [3])
spec = CompositeSpec(action=NdUnboundedContinuousTensorSpec(4), loc=None, scale=None)
net = NormalParamWrapper(torch.nn.GRUCell(4, 8))
module = SafeModule(net, in_keys=["input", "hidden"], out_keys=["loc", "scale"])
td_module = SafeProbabilisticModule(
    module=module,
    spec=spec,
    dist_in_keys=["loc", "scale"],
    sample_out_key=["action"],
    distribution_class=TanhNormal,
    # error doesn't manifest if return_log_prob=False
    return_log_prob=True,
    # return_log_prob=False,
)
params = make_functional(td_module)

# this is fine
td_module(td, params=params)

# vmap breaks it
params = params.expand(4, *params.shape)
td_vmap = vmap(td_module, (None, 0))(td, params)
print(td_vmap)
```

Reply:
Reinstalling I can run this now without crashing the interpreter, instead I get

```
RuntimeError: batched == nullptr INTERNAL ASSERT FAILED at "/Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/functorch/Interpreter.cpp":95, please report a bug to PyTorch.
```